ui/src/datasets.tsx

```diff
@@ -425,13 +487,40 @@ function Preview(props: PreviewProps) {
               return res.query;
             })(),
             (async () => {
-              // TODO(tjennison): Fetch entity instances and refactor processing
-              // code from criteria/concept.
-              return {};
+              const res = await api.searchEntityInstances({
+                entityName: entity.name,
+                underlayName: underlay.name,
+                searchEntityInstancesRequest: {
+                  entityDataset,
+                },
+              });
+
+              const data: TreeGridData = {
+                root: { data: {}, children: [] },
+              };
+              res?.instances?.slice(0, 100)?.forEach((instance, i) => {
```

Comment:
Presumably this slice is only here because we're currently returning way too much data. Should there be some kind of TODO here around setting a maxResults  / pageSizeabove instead?